File: maxSubArray.py
class Solution:
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_sum = nums[0]
        curr_sum = nums[0]
        for i0 in range(1, len(nums)):
            if nums[i0] > curr_sum + nums[i0]:
                curr_sum = nums[i0]
            else:
                curr_sum = curr_sum + nums[i0]
            if curr_sum > max_sum:
                max_sum = curr_sum
        return max_sum

File: java/maxSubArray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        int currSum = nums[0];
        int currIter = 1;
        
        while (currIter < nums.length) {
            currSum = currSum + nums[currIter];
            
            if (nums[currIter] >= currSum)
                currSum = nums[currIter];
            
            if (currSum > maxSum) 
                maxSum = currSum;
            
            currIter++;
        }
        
        return maxSum;
    }
}
